src/compiler/checker.ts

```diff
@@ -6209,7 +6210,13 @@ namespace ts {
             return tupleTypes[arity] || (tupleTypes[arity] = createTupleTypeOfArity(arity));
         }
 
-        function createTupleType(elementTypes: Type[]) {
+        function createTupleType(elementTypes: Type[]): TypeReference {
+            // We have to ensure that we get same instance for empty tuple type,

```

Comment:
`getTupleTypeOfArity` caches the resulting tuple type, so not sure why we need special handling here.